packages/typescript-checker/src/typescript-checker.ts

```diff
@@ -1,201 +1,144 @@
+
+    if (errors.length) {
+      throw new Error(`TypeScript error(s) found in dry run compilation: ${this.formatErrors(errors)}`);
     }
-  }
 
-  private guardTSConfigFileExists() {
-    if (!ts.sys.fileExists(this.tsconfigFile)) {
-      throw new Error(
-        `The tsconfig file does not exist at: "${path.resolve(
-          this.tsconfigFile
-        )}". Please configure the tsconfig file in your stryker.conf file using "${propertyPath<StrykerOptions>('tsconfigFile')}"`
-      );
+    if (Object.keys(dependencyFiles).length === 0) {
+      throw new Error('No sourcefiles were loaded in the compiler.');
     }
+
+    this.sourceFiles = dependencyFiles;
   }
 
-  /**
-   * Checks whether or not a mutant results in a compile error.
-   * Will simply pass through if the file mutated isn't part of the typescript project
-   * @param mutant The mutant to check
-   */
-  public async check(mutant: Mutant): Promise<CheckResult> {
-    if (this.fs.existsInMemory(mutant.fileName)) {
-      this.clearCheckState();
-      this.fs.mutate(mutant);
-      return this.currentTask.promise;
-    } else {
-      // We allow people to mutate files that are not included in this ts project
+  public async check(mutants: MutantTestCoverage[]): Promise<Array<{ mutant: MutantTestCoverage; checkResult: CheckResult }>> {
+    const mutantResults: Array<{ mutant: MutantTestCoverage; errors: ts.Diagnostic[] }> = mutants.map((mutant) => {
       return {
-        status: CheckStatus.Passed,
+        mutant,
+        errors: [],
       };
-    }
-  }
+    });
+
+    // We allow people to mutate files that are not included in this ts project
+    mutants = mutants.filter((mutant) => this.fs.existsInMemory(mutant.fileName));
 
-  /**
-   * Post processes the content of a tsconfig file. Adjusts some options for speed and alters quality options.
-   * @param fileName The tsconfig file name
-   * @param content The tsconfig content
-   * @param buildModeEnabled Whether or not `--build` mode is used
-   */
-  private adjustTSConfigFile(fileName: string, content: string, buildModeEnabled: boolean) {
-    const parsedConfig = ts.parseConfigFileTextToJson(fileName, content);
-    if (parsedConfig.error) {
-      return content; // let the ts compiler deal with this error
-    } else {
-      for (const referencedProject of retrieveReferencedProjects(parsedConfig, path.dirname(fileName))) {
-        this.allTSConfigFiles.add(referencedProject);
+    const errors = await this.typeCheckMutants(mutants);
+    const mutantsToTestIndividual = new Set<Mutant>();
+
+    errors.forEach((error) => {
+      const possibleMutants = this.matchMutantsFromError(mutants, error);
+      if (possibleMutants.length === 1) {
+        mutantResults[mutantResults.findIndex((mutantResult) => mutantResult.mutant.id === possibleMutants[0].id)].errors.push(error);
+      } else if (possibleMutants.length > 1) {
+        possibleMutants.forEach((mutant) => mutantsToTestIndividual.add(mutant));
```

Comment:
You probably want to log on debug level here. Useful information for debugging performance issues later.